config/confighttp/confighttp.go

```diff
@@ -281,6 +281,13 @@ type ServerConfig struct {
 	ResponseHeaders map[string]configopaque.String `mapstructure:"response_headers"`
 }
 
+// NewDefaultServerConfig returns ServerConfig type object with default values.
+// Currently, config options are all initialized as zero values.
+// We encourage to use this function to create an object of ServerConfig.
+func NewDefaultServerConfig() ServerConfig {
+	return ServerConfig{}
```

Comment:
We should initialize the `ResponseHeaders` to prevent users of the API from panics when trying to write to an uninitialized map ([previous comment](https://github.com/open-telemetry/opentelemetry-collector/pull/9672#discussion_r1518358560)).

We should use configtls.NewDefaultServerConfig and configauth.NewDefaultAuthentication and NewDefaultCORSConfig ([previous comment](https://github.com/open-telemetry/opentelemetry-collector/pull/9672/files#r1583251060))

Reply:
nice! OK will add